### PR TITLE
[web] Non-singleton implementation of ui.FlutterView

### PR DESCRIPTION
**Pull request by mdebbar**
- Provide a non-singleton implementation of `EngineFlutterView`.
- Move all common "view" logic into `EngineFlutterView`.
- Remove the unused `windowConfigurations` map from `EnginePlatformDispatcher`.

### CODE REVIEW

**Review by harryterkelsen — APPROVED** [review]
LGTM

**Review by yjbanov — APPROVED** [review]
I have an optional suggestion for an alternative type hierarchy: https://gist.github.com/yjbanov/7948e078f4b41813e2ad5fb1267c3734

LGTM though.

**Comment by mdebbar** [comment]
Went with @yjbanov's suggestion to simplify the type hierarchy.

For future reference: the reason I did the mixin initially was to reuse the code in `ui.*` classes and keep them unchanged. But given that `ui.SingletonFlutterWindow` is deprecated and going away, I think it's okay to let it deviate a bit from the native engine's copy.

File: lib/web_ui/lib/window.dart [hunk 1]
```diff
@@ -29,118 +29,84 @@ abstract class FlutterView {
 }
 
 abstract class SingletonFlutterWindow extends FlutterView {
```
Comment (ditman) [inline lib/web_ui/lib/window.dart:31]: Note: this should be marked as deprecated:

https://github.com/flutter/engine/blob/9cb8599d81669603f0b226a56c25fcd6259b20ed/lib/ui/window.dart#L406-L411

File: lib/web_ui/lib/src/engine/window.dart [hunk 2]
```diff
@@ -38,18 +38,137 @@ const int kImplicitViewId = 0;
 ///
 /// In addition to everything defined in [ui.FlutterView], this class adds
 /// a few web-specific properties.
-abstract interface class EngineFlutterView extends ui.FlutterView {
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/window.dart:41]: Nooo! It was so nice to have an `abstract interface` to code against :P

**Pull request merged**